lib/ansible/modules/cron.py

```diff
@@ -250,8 +251,8 @@ def read(self):
         if self.cron_file:
             # read the cronfile
             try:
-                f = open(self.cron_file, 'r')
-                self.existing = f.read()
+                f = open(self.cron_file, 'rb')
+                self.existing = to_text(f.read(), errors='surrogate_or_strict')
                 self.lines = self.existing.splitlines()
```

Comment:
Convert this to the native type, not text.
```suggestion
                f = open(self.b_cron_file, 'rb')
                self.b_existing = to_native(f.read(), errors='surrogate_or_strict')
```

Ideally, this would be `bytes` and we would work with the data in bytes throughout the module. But there are many places using `self.lines` that need the native text type, so the least invasive change is to convert at the borders: read in the lines as bytes then immediately convert to native text type. Lastly, write the file out as bytes.

Also, we like to prefix native strings with `n_` but that would result in similar ripple effect as using bytes, having to change the variable name throughout the module. I suppose we can omit the name change for now.

Reply:
@samdoran 

> Also, we like to prefix native strings with n_ but that would result in similar ripple effect as using bytes, having to change the variable name throughout the module. I suppose we can omit the name change for now.

So... `self.existing` should be `self.n_existing`, instead of `self.b_existing`, shouldn't be?

I renamed to `self.n_existing`. Please check.

Reply:
Oops, that is correct.